### PR TITLE
Geospatial Preview

### PR DESCRIPTION
**Pull request by max-zilla**
To test this PR:

1) build the previewer container
- Switch to this branch of the `extractors-geo` repo: https://github.com/clowder-framework/extractors-geo/pull/29/files
- In preview.geotiff folder, run `docker build -f Dockerfile -t clowder/extractors-geotiff-preview .`

2) run the geoserver + previewer docker compose
- In clowder2 main directory, run `docker compose -f docker-compose.previewers.yml up -d`

3) upload a geotiff file
- I found one here: https://download.osgeo.org/geotiff/samples/
- The previewer should upload the file to geoserver, then add file metadata to the geotiff.
- Refreshing file view in Clowder should show a zoom to the image, with OSM map behind:
image

UPDATE:
- Need to modify this to look at vis config instead.
- https://github.com/clowder-framework/pyclowder/pull/71 need this to be merged, and update the extractor to write to the config for v2: https://github.com/clowder-framework/extractors-geo/pull/29

### CODE REVIEW

**Comment by sandeep-ps** [comment]
Hi, @max-zilla, For the items in "Update" in the description, are you planning to update this PR, or would that be a new PR?

**Comment by ddey2** [comment]
@max-zilla I see following error after step 2
Screenshot 2023-07-14 at 5 43 46 PM

File: frontend/src/components/visualizations/Geospatial/Geospatial.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+					}),
+				],
+				view: new View({
+					projection: "EPSG:3857",
+					center: center,
+				}),
+				controls: [],
+			});
+			wms_map.getView().fit(bbox);
+			setMap(wms_map);
```
Comment (longshuicy) [inline frontend/src/components/visualizations/Geospatial/Geospatial.tsx:67]: Just curious, your `map` state is never used, why `setMap` is needed?

Comment (longshuicy) [reply frontend/src/components/visualizations/Geospatial/Geospatial.tsx:67]: Ah sorry scratch that. I see you are using the ref inside the wms_map

**Comment by longshuicy** [comment]
> * docker build -f Dockerfile -t clowder/extractors-geotiff-preview .

I made it work by pasting the `geoserver` and `geotiff-preview` image into the docker-compose.dev.yaml; otherwise running the dev yaml and docker-compose.previewers.yml separately doesn't get them onto the same network

File: docker-compose.previewers.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    volumes:
+      - geoserver_data:/opt/geoserver_data
+      - geoserver_exts:/opt/additional_libs
+
+  ## This image must be built from:
+  ##   https://github.com/clowder-framework/extractors-geo/blob/master/preview.geotiff/Dockerfile
+  ## docker build -f Dockerfile -t clowder/extractors-geotiff-preview .
+  ## THIS IS INTENDED FOR LOCAL DEVELOPMENT ONLY.
+  geotiff-preview:
+    image: clowder/extractors-geotiff-preview
```
Comment (longshuicy) [inline docker-compose.previewers.yml:42]: restart: unless-stopped

Comment (longshuicy) [reply docker-compose.previewers.yml:42]: @max-zilla if you add this line in the docker-compose, i can mark your PR as approved. Everything else works for me.

**Review by longshuicy — APPROVED** [review]

**Pull request merged**